pkg/common/common.go

```diff
@@ -113,3 +113,26 @@ func HostnamesToStrings(hostnames []gatewayapiv1alpha2.Hostname) []string {
 	}
 	return hosts
 }
+
+// ValidSubdomains returns (true, "") when every single subdomains item
+// is a subset of at least one of the domains.
+// Domains and subdomains may be prefixed with a wildcard label (*.).
+// The wildcard label must appear by itself as the first label.
+// When one of the subdomains is not a subset of any of the domains, it returns false and
+// the subdomain not being subset of any of the domains
+func ValidSubdomains(domains, subdomains []string) (bool, string) {
+	for _, subdomain := range subdomains {
+		validSubdomain := false
+		for _, domain := range domains {
+			if Name(subdomain).SubsetOf(Name(domain)) {
+				validSubdomain = true
+				break
+			}
+		}
+
+		if !validSubdomain {
+			return false, subdomain
```

Comment:
I understand this is a fail-fast strategy. Could it become annoying for long lists of host names containing multiple invalid ones? Perhaps an enhancement for the future.

Reply:
Indeed, enhancement for the future.